internal/pkg/composable/providers/kubernetes/hints.go

```diff
@@ -256,17 +256,48 @@ func GetHintsMapping(k8sMapping map[string]interface{}, logger *logp.Logger, pre
 		composableMapping: mapstr.M{},
 		processors:        []mapstr.M{},
 	}
+	var hints mapstr.M
+	var containerProcessors []mapstr.M
 
 	if ann, ok := k8sMapping["annotations"]; ok {
 		annotations, _ := ann.(mapstr.M)
-		hints := utils.GenerateHints(annotations, "", prefix)
-		if len(hints) > 0 {
-			logger.Debugf("Extracted hints are :%v", hints)
-			hintData.composableMapping = GenerateHintsMapping(hints, k8sMapping, logger, cID)
-			logger.Debugf("Generated hints mappings are :%v", hintData.composableMapping)
-
-			hintData.processors = utils.GetConfigs(annotations, prefix, processorhints)
-			logger.Debugf("Generated Processors are :%v", hintData.processors)
+
+		if containerEntries, err := annotations.GetValue(prefix + ".hints"); err == nil {
```

Comment:
Since the only actual change here is the input of the `hints := utils.GenerateHints(annotations, "", prefix)`  could you split this into a standalone function so as to isolate the functionality and provide better readability? Also adding some tests would help here.

Reply:
Split in two functions and added a new test `TestGenerateHintsMappingWithProcessorsForContainer` that calls them.

Also made the fixes proposed above. @ChrsMark let me know if you see some else. 

For the documentation update, I opened https://github.com/elastic/ingest-docs/pull/503

As a leftover I need to make some more tests with some other templates other than logs